datahub-frontend/app/controllers/AuthenticationController.java

```diff
@@ -79,13 +81,34 @@ public class AuthenticationController extends Controller {
 
   @Inject
   public AuthenticationController(@Nonnull Config configs) {
+    this.config = configs;
     cookieConfigs = new CookieConfigs(configs);
     jaasConfigs = new JAASConfigs(configs);
     nativeAuthenticationConfigs = new NativeAuthenticationConfigs(configs);
     guestAuthenticationConfigs = new GuestAuthenticationConfigs(configs);
     verbose = configs.hasPath(AUTH_VERBOSE_LOGGING) && configs.getBoolean(AUTH_VERBOSE_LOGGING);
   }
 
+  /**
+   * Gets the configured base path for DataHub.
+   *
+   * @return the normalized base path
+   */
+  @Nonnull
+  private String getBasePath() {
+    return BasePathUtils.normalizeBasePath(config.getString("datahub.basePath"));
```

Comment:
Can this be computed once in the constructor?

Reply:
yes. I think those part of a bunch of places where I left things with debug that can be removed.